lib/router_constraints.rb

```diff
@@ -130,9 +130,18 @@ def matches?(request, path = nil)
 
 class RefIRIRouterConstraint < IRIRouterConstraint
   def matches?(request)
+    regex = %r{
+      (\?[^=]+$) # matches a lone MMT-argument
+      |
+      \?([^=;&]+)[&;].*$ # splits a lone MMT-arg from normal query-string
+      |
+      \?.*$ # drops a the whole standard query-string
+    }x
     # remove the ref/:version_number portion from path
     path = Journey::Router::Utils.unescape_uri(request.original_fullpath).
-      sub(%r{\A/ref/\d+/}, '')
+      sub(%r{\A/ref/\d+/}, '').
+      sub(regex, '\1')
```

Comment:
You're right, it shouldn't be necessary anymore. Do we want to support access tokens with MMT-style URIs?